estate/models/estate_property_offer.py

```diff
@@ -46,6 +46,21 @@ def _inverse_date_deadline(self):
                 offer.date_deadline - (offer.create_date.date() or fields.Date.today())
             ).days
 
+    @api.model
+    def create(self, offers_list):
+        for offer in offers_list:
+            linked_property = self.env['estate.property'].browse(offer['property_id'])
```

Comment:
```suggestion
            linked_property = self.env['estate.property'].browse(offer.get('property_id'))
```
When fetching something from a dictionary we tend to use get to avoid KeyError if the key doesn't exist. I know mostly the property_id will always be there but it is better safe than sorry :sweat_smile:

Reply:
Just learn a new python thing ty 😂

Reply:
By doing that I got this error:
<img width="1002" height="286" alt="image" src="https://github.com/user-attachments/assets/9e975150-e877-445a-bf82-de7eebe7b59b" />

I would do that to fix it :
```py
property_id = offer.get('property_id')
if not property_id:
    continue

linked_property = self.env['estate.property'].browse(property_id)
```

I just think I should raise an error instead of continuing but idk which type of error, I didn't saw any `InternalError` or something similar in the exceptions definitions. What would you do ?

Reply:
Yup, my bad. When using get() and the key doesn't exist it will return None which might cause an Error as browse() is waiting for `int`

what you can do is fallback on a default value in get() like this:

```py
offer.get('property_id', 0)
```

Reply:
But then if their is a property with an id of 0 it will not raise an error but change the wrong property no ?

Reply:
Normally we don't have a record with id 0, our id sequences are 1-indexed. But to stick with the better-safe-than-sorry ideology we can do something like

```py
if offer.get('property_id'):
    linked_property = self.env['estate.property'].browse(offer['property_id'])
```